protos/follow_me/follow_me.proto

```diff
@@ -30,19 +30,42 @@ service FollowMeService {
+      * 
+      * It is only used for setting the follow angle indirectly, for legacy reasons
+      * (since PX4 Follow Me used to use FollowDirection enums for setting, but now
+      * uses the follow_angle_deg float value).
+      * 
+      * DO NOT USE if possible!
+    */
     enum FollowDirection {
-        FOLLOW_DIRECTION_NONE = 0; // Do not follow
+        FOLLOW_DIRECTION_NONE = 0; // Follow direction invalid (defaults to Behind)
         FOLLOW_DIRECTION_BEHIND = 1; // Follow from behind
         FOLLOW_DIRECTION_FRONT = 2; // Follow from front
         FOLLOW_DIRECTION_FRONT_RIGHT = 3; // Follow from front right
         FOLLOW_DIRECTION_FRONT_LEFT = 4; // Follow from front left
+        FOLLOW_DIRECTION_RIGHT = 5; // Follow from right
+        FOLLOW_DIRECTION_LEFT = 6; // Follow from left
+        FOLLOW_DIRECTION_BEHIND_RIGHT = 7; // Follow from behind right
+        FOLLOW_DIRECTION_BEHIND_LEFT = 8; // Follow from behind left
     }
 
-    float min_height_m = 1 [(mavsdk.options.default_value)="8.0"]; // Minimum height for the vehicle in meters (recommended minimum 8 meters)
-    float follow_distance_m = 2 [(mavsdk.options.default_value)="8.0"]; // Distance from target for vehicle to follow in meters (recommended minimum 1 meter)
-    FollowDirection follow_direction = 3; // Direction to follow in
-    float responsiveness = 4 [(mavsdk.options.default_value)="0.5"]; // How responsive the vehicle is to the motion of the target (range 0.0 to 1.0)
+    // Altitude mode to configure which altitude the follow me will assume the target to be at.
+    enum FollowAltitudeMode {
+        FOLLOW_ALTITUDE_MODE_CONSTANT = 0; // Target assumed to be mobing at a constant altitude of home position (where the vehicle armed)
+        FOLLOW_ALTITUDE_MODE_TERRAIN = 1; // Target assumed to be at the terrain level sensed by the distance sensor
+        FOLLOW_ALTITUDE_MODE_TARGET_GPS = 2; // Target GPS altitude taken into account to do 3D tracking
+    }
+
+    float min_height_m = 1 [(mavsdk.options.default_value)="8.0"]; // Follow height in meters (recommended minimum 8 meters)
+    float follow_distance_m = 2 [(mavsdk.options.default_value)="8.0"]; // Follow distance to target in meters (recommended minimum 4 meter)
+    FollowDirection follow_direction = 3; // [!DEPRECIATED!] Direction to follow target at. Exists as a legacy to old PX4 Follow-Me, but now only used for setting the follow_angle_deg. This value never gets sent to the PX4 side!
+    float responsiveness = 4 [(mavsdk.options.default_value)="0.1"]; // How responsive the vehicle is to the motion of the target, Lower value = More responsive (range 0.0 to 1.0)
+    FollowAltitudeMode altitude_mode = 5; // Follow Altitude control mode
+    float max_tangential_vel = 6 [(mavsdk.options.default_value)="8.0"]; // Maximum orbit tangential velocity relative to the target. Higher value = More aggressive follow angle tracking.
```

Comment:
Just realized: we add the units to the variable name (e.g. [see here](https://github.com/mavlink/MAVSDK-Proto/blob/main/protos/telemetry/telemetry.proto#L546)).

```suggestion
    float max_tangential_vel_m_s = 6 [(mavsdk.options.default_value)="8.0"]; // Maximum orbit tangential velocity relative to the target, in meters per second. Higher value = More aggressive follow angle tracking.
```